chainreg/chainregistry.go

```diff
@@ -648,6 +648,49 @@ func NewChainControl(cfg *Config) (*ChainControl, error) {
 		ChainIO:            cc.ChainIO,
 		DefaultConstraints: channelConstraints,
 		NetParams:          *cfg.ActiveNetParams.Params,
+		CurrentNumAnchorChans: func() (int, error) {
```

Comment:
Hmm, seems suuper specific for this use case...

Also the wallet already has access to the full database here, so don't see why it can't just be implemented as a private method within the wallet itself.

Reply:
mhm, made it a private method on the wallet.